### DISCUSSION TITLE
Failed to load renderer.js

### DISCUSSION THREAD

**Discussion by Barrosy** [nklayman/vue-cli-plugin-electron-builder#1905] [Q&A] [answered]
I tried to [implement a renderer script](https://www.electronjs.org/docs/latest/tutorial/quick-start#bonus-add-functionality-to-your-web-contents) such that the implementation meets the Electron Documention standards. When I add a script to the `index.html` file of my project so that the `src` attribute would be configured as: 

Double dots:
```html
<script src="../src/renderer.js"></script>
```
Single dot:
```html
<script src="./src/renderer.js"></script>
```
Or no dots at all:
```html
<script src="src/renderer.js"></script>
```
The developer console will always return either `GET http://localhost:8080/src/renderer.js net::ERR_ABORTED 404 (Not Found)` or `Failed to load resource: the server responded with a status of 404 (Not Found)`.

Does anyone happen to know how to correctly implement a renderer script for this type of project?

My `index.html` would look something along the lines of:
```html
<!DOCTYPE html>
<html lang="">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width,initial-scale=1.0">
    <link rel="icon" href="<%= BASE_URL %>favicon.ico">
    <title><%= htmlWebpackPlugin.options.title %></title>
  </head>
  <body>
    <noscript>
      <strong>We're sorry but <%= htmlWebpackPlugin.options.title %> doesn't work properly without JavaScript enabled. Please enable it to continue.</strong>
    </noscript>
    <div id="app"></div>
    <!-- built files will be auto injected -->
    <!--How to implement renderer?-->
    <!--<script src="./renderer.js" ></script>-->
    <script src="../src/renderer.js"></script>
  </body>
</html>
```
![image](https://user-images.githubusercontent.com/11783496/209722339-6b431d1e-4216-4f44-b524-e4064f87689b.png)

**Comment by MatthijsBurgh** [comment, marked as answer]
I am sure what you want to do with the renderer script. Are you trying to bypass Vue? Maybe you shouldn't be using this project in that case...

**Reply by Barrosy** [reply]
I am not trying to "bypass" Vue. All I'm trying to do is replicate [Electron's set-up](https://www.electronjs.org/docs/latest/tutorial/quick-start#bonus-add-functionality-to-your-web-contents) (see below quote for reference).

> Because the renderer runs in a normal web environment, you can add a <script> tag right before your index.html file's closing </body> tag to include any arbitrary scripts you want 

The main goal for this would be to control the page the way I want as a developer. I could for example just implement a script tag at the bottom of the page just running a simple `console.log("Log something")` such as:
```html
<html>
 <head>
 </head>
 <body>
  <!-- Some content/Vue app here -->
  <script>
   console.log("Log something")
  </script>
 </body>
</html>
```
However for maintenance this would obviously not be a recommended way. I would rather use the `src` attribute of my script tag to keep my files separated. 

Just to be clear, there is no specific underlying goal behind this question at all though. I just don't want to be limited to only utilizing my front-end scripts to Vue as Vue offers the use of mutliple "app" containers. 

Although perhaps there won't be a big probability for my needs of supporting of said containers, why limit Vue's "features" (if that's the right way to put it)? Maybe in the future my app's demands could change and therefor I would like to prevent annoyances that would limit my space of development.

**Reply by MatthijsBurgh** [reply]
This project mostly combines Vue with electron-builder. So I suggest you to dive into the docs of electron-builder.

**Reply by Barrosy** [reply]
> This project mostly combines Vue with electron-builder. So I suggest you to dive into the docs of electron-builder.

I realised this is a Vue issue and not an Electron Builder issue. Thanks for sending me to the right direction though.
